api-server/tox.ini

```diff
@@ -42,27 +52,21 @@ allowlist_externals =
     git
     rm
     tox
-    rockcraft
 deps =
-    juju~=2.9.0
+    juju<4.0
     pytest
     pytest-operator
     ops
 commands =
-    # build and pack rock
-    rockcraft pack
     # clone related charm
     rm -rf {env:LOCAL_CHARM_DIR}
     git clone --branch {env:CHARM_BRANCH} {env:CHARM_REPO} {env:LOCAL_CHARM_DIR}
     # upload rock to docker and microk8s cache, replace charm's container with local rock reference
     bash -c 'NAME=$(yq eval .name rockcraft.yaml) && \
              VERSION=$(yq eval .version rockcraft.yaml) && \
-             ARCH=$(yq eval ".platforms | keys" rockcraft.yaml | awk -F " " '\''{ print $2 }'\'') && \
-             ROCK="$\{NAME\}_$\{VERSION\}_$\{ARCH\}" && \
-             sudo skopeo --insecure-policy copy oci-archive:$ROCK.rock docker-daemon:$ROCK:$VERSION && \
-             docker save $ROCK > $ROCK.tar && \
-             microk8s ctr image import $ROCK.tar && \
-             yq e -i ".resources.oci-image.upstream-source=\"$ROCK:$VERSION\"" {env:LOCAL_CHARM_DIR}/charms/kfp-api/metadata.yaml'
+             DOCKER_IMAGE=$NAME:$VERSION && \
+             docker save $DOCKER_IMAGE > $DOCKER_IMAGE.tar && \
+             sudo microk8s ctr image import $DOCKER_IMAGE.tar --digests=true && \
+             yq e -i ".resources.oci-image.upstream-source=\"$DOCKER_IMAGE\"" {env:LOCAL_CHARM_DIR}/charms/kfp-api/metadata.yaml'
     # run charm integration test with rock
-    tox -c {env:LOCAL_CHARM_DIR} -e integration
-
+    tox -c {env:LOCAL_CHARM_DIR}/charms/kfp-api -e integration -- --model kubeflow
```

Comment:
similar to the sanity checks, is this going to block publishing?

Reply:
Since they fail, yes it will. Should we add a placeholder instead of running tests that fail? Not sure about this.

Reply:
If so, we should do that for all KFP ROCKs, since bundle integration test fail also for unrelated reasons.

Reply:
yeah neither am I.  we can leave this for now